src/lib/DecodedMessage.ts

```diff
@@ -21,6 +21,7 @@ export enum MessageDeliveryStatus {
 }
 
 export class DecodedMessage<
+  ContentType extends DefaultContentTypes[number] = DefaultContentTypes[number],
```

Comment:
Maybe we could have a type for `DefaultContentTypes[number]` like 
`export type DefaultContentType = DefaultContentTypes[number]`

since we use it a lot